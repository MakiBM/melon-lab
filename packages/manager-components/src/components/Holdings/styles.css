.holdings__table-wrap {
  overflow-x: auto;
}

.holdings__quote-asset {
  display: block;
  min-height: 28px;
}

.holdings :global(.holdings__action-cell) {
  width: 130px;
}
