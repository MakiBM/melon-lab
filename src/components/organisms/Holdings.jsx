import React from "react";
import BigNumber from "bignumber.js";

import { Table } from "semantic-ui-react";

const Holdings = ({ holdings, selectAsset, scrollTo, isReadyToTrade }) => (
  <div id="holdings">
    <h3 className="App-intro">Fund Holdings</h3>
    <Table size="small">
      <Table.Header>
        <Table.Row>
          <Table.HeaderCell>Asset</Table.HeaderCell>
          <Table.HeaderCell textAlign="right">Quantity</Table.HeaderCell>
          <Table.HeaderCell textAlign="right">% of portfolio</Table.HeaderCell>
          <Table.HeaderCell textAlign="right">Price (MLN)</Table.HeaderCell>
          <Table.HeaderCell textAlign="right">Trade</Table.HeaderCell>
        </Table.Row>
      </Table.Header>
      <Table.Body>
        {holdings.map(asset => (
          <Table.Row key={asset.name}>
            <Table.Cell>{asset.name}</Table.Cell>
            <Table.Cell textAlign="right">{asset.balance}</Table.Cell>
            <Table.Cell textAlign="right">{asset.percentage}%</Table.Cell>
            <Table.Cell textAlign="right">{asset.price}</Table.Cell>
            <Table.Cell
              textAlign="right"
              onClick={() => selectAsset(asset.name, isReadyToTrade)}
            >
              {asset.name === "MLN-T" ? (
                <div>‚ù§</div>
              ) : isReadyToTrade ? (
                <div className="interactive">Buy/Sell</div>
              ) : (
                <div className="interactive">See Orderbook</div>
              )}
            </Table.Cell>
          </Table.Row>
        ))}
      </Table.Body>
    </Table>
  </div>
);

export default Holdings;
