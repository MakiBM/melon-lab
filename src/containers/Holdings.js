import { connect } from "react-redux";
import { actions } from "../actions/holdings";
import { actions as appActions } from "../actions/app";
import { actions as orderbookActions } from "../actions/orderbook";
import { actions as recentTradesActions } from "../actions/recentTrades";
import { actions as tradeHelperActions } from "../actions/tradeHelper";

import { lifecycle } from "recompose";
import Holdings from "../components/organisms/Holdings";
import displayNumber from "../utils/displayNumber";

const mapStateToProps = state => ({
  holdings: state.holdings.holdings.map(asset => ({
    name: asset.name,
    balance: displayNumber(asset.balance),
    price: displayNumber(asset.price),
    percentage: displayNumber(
      asset.balance
        .times(asset.price)
        .div(state.fund.nav)
        .times(100),
    ),
  })),
});

const mapDispatchToProps = dispatch => ({
  getHoldings: fundAddress => {
    dispatch(actions.getHoldings(fundAddress));
  },
  selectAsset: asset => {
    if (asset !== "MLN-T") {
      dispatch(appActions.updateAssetPair({ base: asset, quote: "MLN-T" }));
      // Here need to reload orderbook, recent trades and trade helper data
      dispatch(orderbookActions.getOrderbook());
      dispatch(recentTradesActions.getRecentTrades());
      // dispatch(tradeHelperActions.tradeInfoRequested());
    }
  },
});

const HoldingsLifecycle = lifecycle({
  componentDidMount() {
    this.props.getHoldings(this.props.fundAddress);
  },
})(Holdings);

const HoldingsContainter = connect(mapStateToProps, mapDispatchToProps)(
  HoldingsLifecycle,
);

export default HoldingsContainter;
