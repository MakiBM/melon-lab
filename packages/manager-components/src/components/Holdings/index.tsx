import React, { Fragment, StatelessComponent } from 'react';
import Button from '~/blocks/Button';
import {
  CellBody,
  CellHead,
  Row,
  Table,
  TableBody,
  TableHead,
} from '~/blocks/Table';

import styles from './styles.css';

export interface HoldingsProps {
  holdings?: any;
  selectAsset?: any;
  isReadyToTrade?: boolean;
  dataValid?: boolean;
  quoteAsset?: any;
}

export const Holdings: StatelessComponent<HoldingsProps> = ({
  holdings,
  selectAsset,
  isReadyToTrade,
  dataValid,
  quoteAsset,
}) => {
  const onClick = (e, assetName) => {
    selectAsset(assetName, isReadyToTrade, quoteAsset);
  };

  return (
    <div className="holdings">
      <style jsx>{styles}</style>
      <h3>Fund Holdings</h3>
      <div className="holdings__table-wrap">
        <Table>
          <TableHead>
            <Row isHead={true}>
              <CellHead>Asset</CellHead>
              <CellHead>Quantity</CellHead>
              <CellHead>% of portfolio</CellHead>
              <CellHead>Price (MLN)</CellHead>
              <CellHead />
            </Row>
          </TableHead>
          <TableBody>
            {holdings && holdings.map(asset => (
              <Row key={asset.name} size="small">
                <CellBody>{asset.name}</CellBody>
                <CellBody>{asset.balance}</CellBody>
                <CellBody>
                  {dataValid && <Fragment>{asset.percentage}</Fragment>}
                </CellBody>
                <CellBody>
                  {dataValid && <Fragment>{asset.price}</Fragment>}
                </CellBody>
                <CellBody cellClass="holdings__action-cell">
                  {asset.name === quoteAsset ? (
                    <span className="holdings__quote-asset" />
                  ) : asset.name !== quoteAsset &&
                  isReadyToTrade &&
                  dataValid ? (
                    <Button
                      size="small"
                      buttonValue={asset.name}
                      onClick={onClick}
                    >
                      Buy/Sell
                    </Button>
                  ) : (
                    <Button
                      size="small"
                      buttonValue={asset.name}
                      onClick={onClick}
                    >
                      See Orderbook
                    </Button>
                  )}
                </CellBody>
              </Row>
            ))}
          </TableBody>
        </Table>
      </div>
    </div>
  );
};

export default Holdings;
