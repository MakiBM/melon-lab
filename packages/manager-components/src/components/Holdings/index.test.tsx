import React from 'react';
import Holdings from './index';

const data = {

};

describe('RecentTrades', () => {
  const defaultElement = <Holdings {...data} />;

  it('should render correctly', () => {
    const wrapper = shallow(defaultElement);
    expect(wrapper).toMatchSnapshot();
  });
});
