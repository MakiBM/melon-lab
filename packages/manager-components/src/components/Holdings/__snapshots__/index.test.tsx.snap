// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Holdings should render correctly 1`] = `
<div
  className="jsx-__scopedHash holdings"
>
  <JSXStyle
    css="__scoped"
    styleId="__scopedHash"
  />
  <h3
    className="jsx-__scopedHash"
  >
    Fund Holdings
  </h3>
  <div
    className="jsx-__scopedHash holdings__table-wrap"
  >
    <Table>
      <TableHead>
        <Row
          isHead={true}
        >
          <CellHead>
            Asset
          </CellHead>
          <CellHead>
            Quantity
          </CellHead>
          <CellHead>
            % of portfolio
          </CellHead>
          <CellHead>
            Price (MLN)
          </CellHead>
          <CellHead />
        </Row>
      </TableHead>
      <TableBody>
        <Row
          key="ANT-T"
        >
          <CellBody>
            ANT-T
          </CellBody>
          <CellBody>
            0.0000
          </CellBody>
          <CellBody>
            0.0000
          </CellBody>
          <CellBody>
            0.0035
          </CellBody>
          <CellBody
            cellClass="holdings__action-cell"
          >
            <Button
              buttonValue="ANT-T"
              onClick={[Function]}
              size="small"
            >
              Buy/Sell
            </Button>
          </CellBody>
        </Row>
        <Row
          key="BAT-T"
        >
          <CellBody>
            BAT-T
          </CellBody>
          <CellBody>
            0.0000
          </CellBody>
          <CellBody>
            0.0000
          </CellBody>
          <CellBody>
            0.0007
          </CellBody>
          <CellBody
            cellClass="holdings__action-cell"
          >
            <Button
              buttonValue="BAT-T"
              onClick={[Function]}
              size="small"
            >
              Buy/Sell
            </Button>
          </CellBody>
        </Row>
        <Row
          key="DGD-T"
        >
          <CellBody>
            DGD-T
          </CellBody>
          <CellBody>
            0.0000
          </CellBody>
          <CellBody>
            0.0000
          </CellBody>
          <CellBody>
            0.0000
          </CellBody>
          <CellBody
            cellClass="holdings__action-cell"
          >
            <Button
              buttonValue="DGD-T"
              onClick={[Function]}
              size="small"
            >
              Buy/Sell
            </Button>
          </CellBody>
        </Row>
        <Row
          key="DGX-T"
        >
          <CellBody>
            DGX-T
          </CellBody>
          <CellBody>
            0.0000
          </CellBody>
          <CellBody>
            0.0000
          </CellBody>
          <CellBody>
            0.0000
          </CellBody>
          <CellBody
            cellClass="holdings__action-cell"
          >
            <Button
              buttonValue="DGX-T"
              onClick={[Function]}
              size="small"
            >
              Buy/Sell
            </Button>
          </CellBody>
        </Row>
        <Row
          key="GNO-T"
        >
          <CellBody>
            GNO-T
          </CellBody>
          <CellBody>
            0.0000
          </CellBody>
          <CellBody>
            0.0000
          </CellBody>
          <CellBody>
            0.1113
          </CellBody>
          <CellBody
            cellClass="holdings__action-cell"
          >
            <Button
              buttonValue="GNO-T"
              onClick={[Function]}
              size="small"
            >
              Buy/Sell
            </Button>
          </CellBody>
        </Row>
        <Row
          key="REP-T"
        >
          <CellBody>
            REP-T
          </CellBody>
          <CellBody>
            0.0000
          </CellBody>
          <CellBody>
            0.0000
          </CellBody>
          <CellBody>
            0.0648
          </CellBody>
          <CellBody
            cellClass="holdings__action-cell"
          >
            <Button
              buttonValue="REP-T"
              onClick={[Function]}
              size="small"
            >
              Buy/Sell
            </Button>
          </CellBody>
        </Row>
        <Row
          key="ZRX-T"
        >
          <CellBody>
            ZRX-T
          </CellBody>
          <CellBody>
            0.0000
          </CellBody>
          <CellBody>
            0.0000
          </CellBody>
          <CellBody>
            0.0024
          </CellBody>
          <CellBody
            cellClass="holdings__action-cell"
          >
            <Button
              buttonValue="ZRX-T"
              onClick={[Function]}
              size="small"
            >
              Buy/Sell
            </Button>
          </CellBody>
        </Row>
        <Row
          key="REQ-T"
        >
          <CellBody>
            REQ-T
          </CellBody>
          <CellBody>
            0.0000
          </CellBody>
          <CellBody>
            0.0000
          </CellBody>
          <CellBody>
            0.0001
          </CellBody>
          <CellBody
            cellClass="holdings__action-cell"
          >
            <Button
              buttonValue="REQ-T"
              onClick={[Function]}
              size="small"
            >
              Buy/Sell
            </Button>
          </CellBody>
        </Row>
        <Row
          key="WETH-T"
        >
          <CellBody>
            WETH-T
          </CellBody>
          <CellBody>
            0.5166
          </CellBody>
          <CellBody>
            50.5764
          </CellBody>
          <CellBody>
            1.0000
          </CellBody>
          <CellBody
            cellClass="holdings__action-cell"
          >
            <span
              className="jsx-__scopedHash holdings__quote-asset"
            />
          </CellBody>
        </Row>
        <Row
          key="MLN-T"
        >
          <CellBody>
            MLN-T
          </CellBody>
          <CellBody>
            10.7338
          </CellBody>
          <CellBody>
            49.4236
          </CellBody>
          <CellBody>
            0.0470
          </CellBody>
          <CellBody
            cellClass="holdings__action-cell"
          >
            <Button
              buttonValue="MLN-T"
              onClick={[Function]}
              size="small"
            >
              Buy/Sell
            </Button>
          </CellBody>
        </Row>
        <Row
          key="MKR-T"
        >
          <CellBody>
            MKR-T
          </CellBody>
          <CellBody>
            0.0000
          </CellBody>
          <CellBody>
            0.0000
          </CellBody>
          <CellBody>
            1.3430
          </CellBody>
          <CellBody
            cellClass="holdings__action-cell"
          >
            <Button
              buttonValue="MKR-T"
              onClick={[Function]}
              size="small"
            >
              Buy/Sell
            </Button>
          </CellBody>
        </Row>
        <Row
          key="DAI-T"
        >
          <CellBody>
            DAI-T
          </CellBody>
          <CellBody>
            0.0000
          </CellBody>
          <CellBody>
            0.0000
          </CellBody>
          <CellBody>
            0.0021
          </CellBody>
          <CellBody
            cellClass="holdings__action-cell"
          >
            <Button
              buttonValue="DAI-T"
              onClick={[Function]}
              size="small"
            >
              Buy/Sell
            </Button>
          </CellBody>
        </Row>
        <Row
          key="KNC-T"
        >
          <CellBody>
            KNC-T
          </CellBody>
          <CellBody>
            0.0000
          </CellBody>
          <CellBody>
            0.0000
          </CellBody>
          <CellBody>
            0.0276
          </CellBody>
          <CellBody
            cellClass="holdings__action-cell"
          >
            <Button
              buttonValue="KNC-T"
              onClick={[Function]}
              size="small"
            >
              Buy/Sell
            </Button>
          </CellBody>
        </Row>
        <Row
          key="JNT-T"
        >
          <CellBody>
            JNT-T
          </CellBody>
          <CellBody>
            0.0000
          </CellBody>
          <CellBody>
            0.0000
          </CellBody>
          <CellBody>
            0.0003
          </CellBody>
          <CellBody
            cellClass="holdings__action-cell"
          >
            <Button
              buttonValue="JNT-T"
              onClick={[Function]}
              size="small"
            >
              Buy/Sell
            </Button>
          </CellBody>
        </Row>
        <Row
          key="OMG-T"
        >
          <CellBody>
            OMG-T
          </CellBody>
          <CellBody>
            0.0000
          </CellBody>
          <CellBody>
            0.0000
          </CellBody>
          <CellBody>
            0.0150
          </CellBody>
          <CellBody
            cellClass="holdings__action-cell"
          >
            <Button
              buttonValue="OMG-T"
              onClick={[Function]}
              size="small"
            >
              Buy/Sell
            </Button>
          </CellBody>
        </Row>
        <Row
          key="NMR-T"
        >
          <CellBody>
            NMR-T
          </CellBody>
          <CellBody>
            0.0000
          </CellBody>
          <CellBody>
            0.0000
          </CellBody>
          <CellBody>
            0.0184
          </CellBody>
          <CellBody
            cellClass="holdings__action-cell"
          >
            <Button
              buttonValue="NMR-T"
              onClick={[Function]}
              size="small"
            >
              Buy/Sell
            </Button>
          </CellBody>
        </Row>
      </TableBody>
    </Table>
  </div>
</div>
`;
